app/services/login/user_verifier.rb

```diff
@@ -134,7 +136,11 @@ def attempt_secondary_idme_identifier
       @identifier = idme_uuid
       raise Errors::UserVerificationNotCreatedError if identifier.nil?
 
-      Rails.logger.info("[Login::UserVerifier] Attempting alternate type=#{type}  identifier=#{identifier}")
+      Rails.logger.info("[Login::UserVerifier] Attempting alternate type=#{type} identifier=#{identifier}")
+    end
+
+    def locked
+      @locked ||= existing_user_account&.user_verifications&.send(login_type)&.where(locked: true).present?
```

Comment:
Can you start with function with `return false unless existing_user_account`?

This way you can remove all of the safety `&.` operators here

Reply:
I removed it from the user_account & user_verification since we know they'll be there with the guard clause, but the last one I think needs to stay because most times the existing account won't have a user_verification of the CSP type so it will be trying to call `.where` on a `nil` value.

Reply:
actually, you don't have to worry about `nil`, this is the beauty of ActiveRecord chaining:

> UserAccount.last.user_verifications.send(:dslogon)
=> []

And then if you call `.where` on the result of this:
> UserAccount.last.user_verifications.send(:dslogon).where(locked: true)
=> []

How is that possible? It's because ActiveRecord queries actually return an `ActiveRecord_AssociationRelation`, which you can still chain queries off of even if it's empty:
> UserAccount.last.user_verifications.send(:dslogon).class
=> UserVerification::ActiveRecord_AssociationRelation